### PR TITLE
9270 bump go to 1.17 in 1.3

### LINKED ISSUE #9270: Update Go client to use 1.17
**Description**

Our Go client is now built against 1.15, which is not supported anymore. We should start building against Go 1.17, which is the lowest support version.

### PR DESCRIPTION
**Pull request by pihme**
## Description
- Bumps Go to version 1.17

**Review Hints**

- Does not remove deprecated code (there will be follow up issues for that)
## Related issues

closes #9270

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     steps:
       - uses: actions/setup-go@v3
         with:
-          go-version: '~1.15'
+          go-version: 1.17
```
Comment (npepinpe) [inline .github/workflows/golangci-lint.yml:16]: :question: Is '1.17' equivalent to `~1.17` (which AFAIK will pick up the latest patch version on the 1.17 version)?

**Review by npepinpe — APPROVED** [review]
:+1:

**Comment by npepinpe** [comment]
Sorry, I just realized we forgot to update the release scripts to use Go 1.17. This is done in [.ci/pipelines/release_zeebe.groovy](https://github.com/camunda/zeebe/blob/001f16b65c4d2e5969fb602583bee49ed55336d5/.ci/pipelines/release_zeebe.groovy)

**Review by npepinpe — CHANGES_REQUESTED** [review]
Sorry, I just realized we forgot to update the release scripts to use Go 1.17. This is done in [.ci/pipelines/release_zeebe.groovy](https://github.com/camunda/zeebe/blob/001f16b65c4d2e5969fb602583bee49ed55336d5/.ci/pipelines/release_zeebe.groovy). This would have to be done on the main branch as well.

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (npepinpe) [reply .github/workflows/golangci-lint.yml:16]: It's probably OK to have any 1.17 as Go follows semver, so we wouldn't be relying on any specific patch version I think for just testing, only for releasing. That said, getting it from the gomod file is probably the most portable for the future as well, if it's not too complicated to do :+1:

Comment (pihme) [reply .github/workflows/golangci-lint.yml:16]: I'll probably leave it as is here and then try to improve it on main.

**Review by npepinpe — APPROVED** [review]
:+1:

**Comment by pihme** [comment]
bors merge

**Pull request merged**